### ISSUE TITLE
Unexpected on_block event after websocket close

### ISSUE THREAD

**Issue by bitfag** [Codaone/DEXBot#257]
To reproduce: define meaningful `self.ontick += myfunc` method for the worker. Then stop dexbot by C-c and you will see `stop()` than `myfunc()`.

The reason of this: `on_block` event comes from Notify instance even after `self.notify.websocket.close()`.

Here is the pdb trace showing this:

```
2018-08-02 14:31:27,258 (staggered_sell:343) - dtest1 - INFO - Done orders placement
^C> /home/vvk/devel/DEXBot/dexbot/worker.py(188)stop()
-> if worker_name and len(self.workers) > 1:
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(202)stop()
-> if pause:
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(203)stop()
-> for worker in self.workers:
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(204)stop()
-> self.workers[worker].pause()
(Pdb) n
2018-08-02 14:32:24,107 (basestrategy:401) - dtest1 - INFO - Canceling all orders
2018-08-02 14:32:29,333 (basestrategy:404) - dtest1 - INFO - Orders canceled
> /home/vvk/devel/DEXBot/dexbot/worker.py(203)stop()
-> for worker in self.workers:
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(205)stop()
-> if self.notify:
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(206)stop()
-> self.notify.websocket.close()
(Pdb) n
close status: 31522
--Return--
> /home/vvk/devel/DEXBot/dexbot/worker.py(206)stop()->None
-> self.notify.websocket.close()
(Pdb) n
--Return--
> /home/vvk/devel/DEXBot/dexbot/cli.py(83)<lambda>()->None
-> kill_workers = worker_job(worker, lambda: worker.stop(pause=True))
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(112)on_block()
-> for job in self.jobs:
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(115)on_block()
-> self.jobs = set()
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(117)on_block()
-> self.config_lock.acquire()
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(118)on_block()
-> for worker_name, worker in self.config["workers"].items():
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(119)on_block()
-> if worker_name not in self.workers or self.workers[worker_name].disabled:
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(121)on_block()
-> try:
(Pdb) n
> /home/vvk/devel/DEXBot/dexbot/worker.py(122)on_block()
-> self.workers[worker_name].ontick(data)
(Pdb) n
2018-08-02 14:34:15,587 (basestrategy:401) - dtest1 - INFO - Canceling all orders
Connection is already closed.
-1
2018-08-02 14:34:16,182 (basestrategy:404) - dtest1 - INFO - Orders canceled
2018-08-02 14:34:16,466 (staggered_sell:244) - dtest1 - DEBUG - Number of full orders to place: 4
```

Probably this is actually python-bitshares bug, but we can implement a quick solution for this (will push a PR soon).

**Comment by bitfag** [comment]
OK I failed to fix it dexbot-wide and created a workaround inside the strategy like this:

```python
    def pause(self):
        """ Pause the worker
        """
        self.ontick -= self.on_market_update_wrapper
        self.cancel_all()
        self.clear_orders()
```

**Comment by thehapax** [comment]
@bitfag can we mark this issue as closed?

**Comment by bitfag** [comment]
@thehapax no I don't think the original problem was resolved, I would like to keep this in the backlog queue, maybe this issue will show up later.

**Comment by bitfag** [comment]
I rechecked this issue and was unable to reproduce, because @joelvai touched the related code and fixed c9bfe009f5693c81bbd451b4d4f20e5bbd5557da maybe without intention :)
